### PR TITLE
modulesync 4.0.0-8-g292033c

### PR DESCRIPTION
**Pull request by attachmentgenie**
initial module sync

### CODE REVIEW

**Comment by attachmentgenie** [comment]
#26 is required for this to pass

**Comment by bastelfreak** [comment]
@attachmentgenie can you rebae please? It might be required to convert the CentOS/RHEL versions to integers in metadata.json.

**Comment by bastelfreak** [comment]
@attachmentgenie can you rebase against master to get rid of the merge commits and convert the centos/rhel version from floats to integers in metadata.json?

**Comment by attachmentgenie** [comment]
Sure thing

**Comment by ekohl** [comment]
I added some fixes to hopefully make the tests pass. I also rebased it and force pushed it for a bit cleaner history.

**Comment by ekohl** [comment]
The tests should be converted to rspec-puppet-facts. It's also a good thing to convert params.pp to use the service_provider fact from stdlib.

**Comment by ekohl** [comment]
I pushed a large set of changes that are backwards incompatible. However, it was the easiest way for me to make the module tests pass using modern tooling. I didn't check acceptance tests yet.

**Comment by ekohl** [comment]
The JSON function tests are still broken and it looks like there's a duplicate implementation.

**Comment by ekohl** [comment]
@attachmentgenie it's only the function part that's left. I'll let you deal with it since you made changes there in the past.

**Comment by attachmentgenie** [comment]
> @attachmentgenie it's only the function part that's left. I'll let you deal with it since you made changes there in the past.

@ekohl thanks for the help, ill be having a look at that function

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
-  c.before :suite do
-    # Install module and dependencies
-    puppet_module_install(:source => proj_root, :module_name => 'nomad')
-    hosts.each do |host|
-      on host, puppet('module', 'install', 'puppet-stdlib'), { :acceptable_exit_codes => [0,1] }
-      on host, puppet('module', 'install', 'puppet-archive'), { :acceptable_exit_codes => [0,1] }
-    end
-  end
+configure_beaker do |host|
+  host.install_package('unzip')
```
Comment (bastelfreak) [inline spec/spec_helper_acceptance.rb:4]: lol, didn't know `install_package()` is a thing. neat!

File: data/common.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+nomad::init_style: "%{facts.service_provider}"
```
Comment (bastelfreak) [inline data/common.yaml:2]: can you move this to the init.pp? That allows puppet-strings to pick it up

Comment (attachmentgenie) [reply data/common.yaml]: i actually already did, but forgot to remove it here

**Review by bastelfreak — APPROVED** [review]
this is a bit tricky to read, because a few changed could be pulled out into dedicated PRs,, like puppet-lint fixes to hiera migration, but I think this is good enough to be merged

**Pull request merged**